### PR TITLE
[8.0][account_invoice_kanban] Kanban view for invoice to manage administrative tasks

### PR DESCRIPTION
**Pull request by sudhir-serpentcs**
Added 'account_invoice_kanban' module for kanban view of invoices like customer invoice and supplier invoice.

@elicoidal

### CODE REVIEW

File: account_invoice_kanban/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# © 2017 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'Account Invoice Kanban',
+    'summary': 'Improve the Invoice view',
```
Comment (elicoidal) [inline account_invoice_kanban/__openerp__.py]: Add Kanban view in Invoice to follow administrative tasks

**Review by elicoidal — CHANGES_REQUESTED** [review]
Really good :)
Small details: invoices should be ordered via the due date (older on top)

**Comment by elicoidal** [comment]
Another detail: add the group by "stage"

**Comment by elicoidal** [comment]
Please fix travis and runbot

**Comment by sudhir-serpentcs** [comment]
@elicoidal 
can you please review code.

**Review by elicoidal — APPROVED** [review]

**Comment by elicoidal** [comment]
Can you check Travis and Runbot?
Other than that, LGTM

**Comment by sudhir-serpentcs** [comment]
@elicoidal Thanks for the feedback.
I looked at the error logs of Travis, Runbot and it seems the issue is from other module (stock_picking_invoicing_incoterm)

**Error:**

raise orm.except_orm(_('Error'), _("You try to install module '%s' that depends on module '%s'.\nBut the latter module is not available in your system.") % (module.name, dep.name,))
except_orm: ('Error', u"You try to install module 'stock_picking_invoicing_incoterm' that depends on module 'stock_picking_invoicing'.\nBut the latter module is not available in your system.")

**Comment by elicoidal** [comment]
@sudhir-serpentcs thanks.
Let's wait for other reviewers to start

**Comment by elicoidal** [comment]
@jbeficent @max3903 @fclementic2c can you help with the review?

**Comment by andhit-r** [comment]
Hi @sudhir-serpentcs , thanks for your contribution. I wonder why you don't use project_model_to_task to achieve same purpose?

**Comment by elicoidal** [comment]
project_model_to_task would be able able to answer somehow the requirement indeed but is more general and less integrated to the invoice structure. 
Besides honestly you need to be aware of the possibility while this module is straightforward to use.
If you think there is no interest, we will close and publish in our github
@andhit-r

**Comment by andhit-r** [comment]
@elicoidal i could not know for sure if there will be an interest/no interest for this PR. But here are my opinion:

* If you would like to achieve straightforward usage i would prefer to create own stage object. This approach will not install project module, hence straight forward will be achieved. 
* If your purpose is to give F&A people some kind of task management, i recommend to use the project_model_to_task. This approach will give an opportunity to track progress not only for invoice but all F&A doc.

**Comment by elicoidal** [comment]
@andhit-r Fair enough :)
We will create the stage in the invoicing part and reapply.
cc @sudhir-serpentcs

**Comment by elicoidal** [comment]
@sudhir-serpentcs let me know when you can restart this one

**Comment by elicoidal** [comment]
@sudhir-serpentcs Any feedback on this one?

**Comment by sudhir-serpentcs** [comment]
@elicoidal Applied stages in invoicing, could you please code review?

File: account_invoice_kanban/models/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+class AccountInvoiceStage(models.Model):
+    _name = 'account.invoice.stage'
+    _description = 'Account Invoice Stage'
+
+    name = fields.Char('Stage Name', required=True)
+    description = fields.Char('Description')
+    sequence = fields.Integer('Sequence', default=1)
+    fold = fields.Boolean('Folded in Kanban View')
+    used_in_invoice = fields.Boolean('Used in Invoicing')
```
Comment (elicoidal) [inline account_invoice_kanban/models/account_invoice.py]: not needed anymore

**Review by elicoidal — CHANGES_REQUESTED** [review]
Please check travis and runbot

**Comment by sudhir-serpentcs** [comment]
@elicoidal  Made the change as requested.

But the travis are still red it is failing for some other module, non-proposed module.

**Comment by elicoidal** [comment]
@sudhir-serpentcs Almost perfect.
Menu entry for the invoice stage should depend on the main configuration menu (just before misc entry)
then for me is a good to go

**Comment by elicoidal** [comment]
@andhit-r can you have a look?

**Review by elicoidal — APPROVED** [review]

File: account_invoice_kanban/README.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+* Add a Stage in Invoice (like in tasks) with the separate field for the additional
+  workflow on top of current one.
+* No button to switch from stages: simply clicking on it is enough (like tasks)
+* Add Kanban view, default with stage group by. 
+* Kanban becomes the default view.
+
+Configuration
+=============
+Stage are set up in the Project application (standard tasks stages):
```
Comment (andhit-r) [inline account_invoice_kanban/README.rst]: You should adjust this step according newest change

**Review by andhit-r — APPROVED** [review]
Just minor comment on README. Runbot still red even it's working. Travis complain because the main 8.0 build broken.

**Comment by sudhir-serpentcs** [comment]
Made requested changes, thanks @elicoidal @andhit-r

**Comment by MiquelRForgeFlow** [comment]
@sudhir-serpentcs , Runbot and Travis both have now the same warning:
`WARNING openerp_test openerp.modules.loading: The model account.invoice.stage has no access rules, consider adding one. E.g. access_account_invoice_stage,access_account_invoice_stage,model_account_invoice_stage,,1,0,0,0`

It's seems that the warning is related to the class `account.invoice.stage` that you created.

Please, could you fix that? 🙂

**Comment by sudhir-serpentcs** [comment]
@mreficent thanks for the feedback. I will add the rights to account user and manager for the new object.

**Pull request merged**